task_2/task_2.py

```diff
@@ -0,0 +1,430 @@
+def rerenge_data(data):
+    rerange_gata= {}
+
+    for user in data:
+        key_year = str(int(int(user['dob.date'][8:])/10)*10) + '-th'
+        key_year = '00-th' if key_year == '0-th' else key_year
+
+        key_country = user['location.country']
+        if key_year not in rerange_gata:
+            rerange_gata[key_year] = {}
+        if key_country not in rerange_gata[key_year]:
+            rerange_gata[key_year][key_country] = []
+        rerange_gata[key_year][key_country].append(user)
+    # print(rerange_gata['80-th']['United States'])
+    return rerange_gata
+
+
+# 9.  In the destination folder create sub folders for every decade
+def create_folder_structure(dict_list_dir):
+    change_workdir(Var.output_folder)
+    for decade_dir in dict_list_dir.keys():
+        if not os.path.exists(decade_dir):  # decades
+            os.mkdir(decade_dir)
+            for country in dict_list_dir[decade_dir].keys():  # country
+                path_country = os.path.join(decade_dir, country)
+                if not os.path.exists(path_country):
+                    os.mkdir(path_country)
+
+
+# 11. Store the data to csv files in corresponding folders. Name of file should be composed using
+#     the following pattern: max_age_{}_avg_registered_{}_ popular_id_{}.csv
+#     (max age, average number of years of registration, the most common id.name)
+#     Log paths of created files.
+def fill_data_in_folder_structure(data):
+    for decade in data.keys():
+        for country in data[decade].keys():
+            temp_dict = data[decade][country]
+
+            max_age = max([reg_age['dob.age'] for reg_age in data[decade][country]])
+
+            avg = sum([int(reg_age['registered.age']) for reg_age in data[decade][country]]) / len(data[decade][country])
+
+            id_name_counts = Counter([name['id.name'] for name in data[decade][country]])
+            popular_id = id_name_counts.most_common(1)[0][0]
+
+            file_name = f'max_age_{max_age}_avg_registered_{int(avg)}_popular_id_{popular_id}.csv'
+
+            path_file = os.path.join(Var.output_folder, decade, country, file_name)
+
+            with open(path_file, mode='w', newline='', encoding="utf-8") as file:
+                writer = csv.DictWriter(file, fieldnames=dict(temp_dict[0]).keys())
+                writer.writeheader()
+                writer.writerows([data for data in temp_dict])
+
+            print(path_file)
+
+# 12. Remove the data before 1960-th
+def remove_data_before_decade(data, min_decade):
+    decades = [decade for decade in data.keys()]
+    for decade in decades:
+        for country in data[decade]:
+            if int(data[decade][country][0]['dob.date'][6:]) < min_decade:
+                del data[decade]
+                path = os.path.join(Var.output_folder, decade)
+                print(path)
+                if os.path.exists(path):
+                    shutil.rmtree(path)
+                break
```

Comment:
you don't need the loop by country.
you already have the value of the decade. you don't need to check date of birth.
Also you don't need to check the existence. you created these folders. They will be in place.